author	Shiji Yang <yangshiji66@qq.com>	

Sun, 5 Jan 2025 16:09:01 +0530 (18:39 +0800)

committer	Daniel Golle <daniel@makrotopia.org>	

Tue, 7 Jan 2025 05:17:08 +0530 (23:47 +0000)

commit	1e5c40e07928ab6de57ee1811945944ccb4f1907
 
jail: fix NULL-pointer dereference when connection to ubus failed
 
Exit when parent_ctx is a NULL pointer.
 
Signed-off-by: Shiji Yang <yangshiji66@qq.com>
 

Index: procd-2024.12.22~42d39376/jail/jail.c
===================================================================
--- procd-2024.12.22~42d39376.orig/jail/jail.c	2024-12-22 21:34:11.000000000 +0530
+++ procd-2024.12.22~42d39376/jail/jail.c	2025-09-09 11:15:47.486139337 +0530
@@ -2858,6 +2858,12 @@
 	signals_init();
 
 	parent_ctx = ubus_connect(NULL);
+	if (!parent_ctx) {
+		ERROR("Connection to ubus failed\n");
+		ret = -ECONNREFUSED;
+		goto errout;
+	}
+
 	ubus_add_uloop(parent_ctx);
 
 	if (opts.ocibundle) {
